Mathlib/Order/JordanHolder.lean

```diff
@@ -624,36 +354,37 @@ theorem trans {s₁ s₂ s₃ : CompositionSeries X} (h₁ : Equivalent s₁ s
     fun i => iso_trans (h₁.choose_spec i) (h₂.choose_spec (h₁.choose i))⟩
 #align composition_series.equivalent.trans CompositionSeries.Equivalent.trans
 
-theorem append {s₁ s₂ t₁ t₂ : CompositionSeries X} (hs : s₁.top = s₂.bot) (ht : t₁.top = t₂.bot)
+protected theorem smash {s₁ s₂ t₁ t₂ : CompositionSeries X}
+    (hs : s₁.last = s₂.head) (ht : t₁.last = t₂.head)
     (h₁ : Equivalent s₁ t₁) (h₂ : Equivalent s₂ t₂) :
-    Equivalent (append s₁ s₂ hs) (append t₁ t₂ ht) :=
+    Equivalent (smash s₁ s₂ hs) (smash t₁ t₂ ht) :=
   let e : Fin (s₁.length + s₂.length) ≃ Fin (t₁.length + t₂.length) :=
     calc
       Fin (s₁.length + s₂.length) ≃ Sum (Fin s₁.length) (Fin s₂.length) := finSumFinEquiv.symm
       _ ≃ Sum (Fin t₁.length) (Fin t₂.length) := Equiv.sumCongr h₁.choose h₂.choose
       _ ≃ Fin (t₁.length + t₂.length) := finSumFinEquiv
-
   ⟨e, by
     intro i
-    refine' Fin.addCases _ _ i
+    refine Fin.addCases ?_ ?_ i
     · intro i
-      simpa [e, top, bot] using h₁.choose_spec i
+      simpa [-smash_toFun, e, smash_castAdd, smash_succ_castAdd] using h₁.choose_spec i
```

Comment:
It's a shame that you have to do `-smash_toFun` here. Should it not be a simp lemma in general?

Reply:
yeah, `smash_toFun` shouldn't be a thing. It is removed now